src/init.cpp

```diff
@@ -827,8 +827,10 @@ void InitParameterInteraction(ArgsManager& args)
             LogPrintf("%s: parameter interaction: -listen=0 -> setting -upnp=0\n", __func__);
         if (args.SoftSetBoolArg("-discover", false))
             LogPrintf("%s: parameter interaction: -listen=0 -> setting -discover=0\n", __func__);
-        if (args.SoftSetBoolArg("-listenonion", false))
+        if (args.GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION)) {
+            args.ForceSetArg("-listenonion", std::string("0"));
```

Comment:
I think refusing to start (with an error message?) over changing what the user has set is better

Reply:
Agree with you guys. Will update the PR to refuse the start instead of changing the explicitly set parameter.

Reply:
> I think refusing to start (with an error message?) over changing what the user has set is better

I've sent a new PR with this behavior. Refusing to start looks more UNIXish.

Reply:
> The rest of `InitParameterInteraction()` uses `SoftSetBoolArg()` which I guess is to avoid overriding a value that has been explicitly set by the user. `InitParameterInteraction()` only flips defaults. I think this behavior is sensible as it minimizes the user's surprise.
> 
> There is some [other code](https://github.com/bitcoin/bitcoin/blob/ddbf7a6f5051b6d7faed54828445d6eaadcd8441/src/init.cpp#L1011-L1024) which refuses to start the application if incompatible options are given - for example `-prune=1 -txindex=1` or `-bind=addr:port -listen=0`.
> 
> Given the above, what about refusing to start if `-listen=0 -listenonion=1` is explicitly given?

Thanks for the idea. I've updated the PR with this behavior.